### PR TITLE
Add Service Log integration

### PR DESCRIPTION
**Pull request by JiriPapousek**
# Description

Adding connection to content template render service. The service is used to interpolate rule description and reason templates with data from reports and the result is sent to the Service Log.

Deployment of the content template renderer service is necessary for the functionality of proposed changes.

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing steps

The functionality was tested with locally deployed external data pipeline and debugging prints, as well as with modified tests.

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: differ/differ.go [hunk 1]
```diff
@@ -319,21 +345,34 @@ func processReportsByCluster(ruleContent types.RulesMap, storage Storage, cluste
 			string(cluster.ClusterName))
 		notifiedAt := types.Timestamp(time.Now())
 
+		renderedReports, err := renderReportsForCluster(conf.GetDependenciesConfiguration(config), cluster.ClusterName, deserialized.Reports, ruleContent)
```
Comment (tisnik) [inline differ/differ.go:348]: we need this only for ServiceLog, ask @dpensi how to distinguish please

File: differ/differ.go [hunk 2]
```diff
@@ -355,7 +395,7 @@ func processReportsByCluster(ruleContent types.RulesMap, storage Storage, cluste
 				log.Info().Str(clusterName, string(cluster.ClusterName)).Msg("Different report from the last one")
 				ReportWithHighImpact.Inc()
 				notificationPayloadURL := generateNotificationPayloadURL(&notificationRuleDetailsURI, string(cluster.ClusterName), module, errorKey)
-				appendEventToNotificationMessage(notificationPayloadURL, &notificationMsg, description, totalRisk, time.Time(cluster.UpdatedAt).UTC().Format(time.RFC3339Nano))
+				appendEventToNotificationMessage(notificationPayloadURL, &notificationMsg, renderedReport.Description, totalRisk, time.Time(cluster.UpdatedAt).UTC().Format(time.RFC3339Nano))
```
Comment (tisnik) [inline differ/differ.go:398]: only for ServiceLog!

File: differ/differ.go [hunk 3]
```diff
@@ -438,17 +478,31 @@ func processAllReportsFromCurrentWeek(ruleContent types.RulesMap, storage Storag
 		}
 		digest.Recommendations += numReports
 
+		renderedReports, err := renderReportsForCluster(conf.GetDependenciesConfiguration(config), cluster.ClusterName, deserialized.Reports, ruleContent)
```
Comment (tisnik) [inline differ/differ.go:481]: ServiceLog don't have weekly status

**Review by tisnik — COMMENTED** [review]
Some comments

File: differ/differ.go [hunk 4]
```diff
@@ -368,7 +477,12 @@ func processReportsByCluster(ruleContent types.RulesMap, storage Storage, cluste
 		}
 
 		log.Info().Msgf("Producing instant notification for cluster %s with %d events", string(cluster.ClusterName), len(notificationMsg.Events))
-		_, _, err = notifier.ProduceMessage(notificationMsg)
+		msgBytes, err := json.Marshal(notificationMsg)
+		if err != nil {
+			log.Error().Err(err).Msg(invalidJSONContent)
+			os.Exit(ExitStatusMessageEncoderError)
```
Comment (tisnik) [inline differ/differ.go:483]: so it exit even because one wrong message?

File: differ/differ.go [hunk 5]
```diff
@@ -476,7 +590,12 @@ func processAllReportsFromCurrentWeek(ruleContent types.RulesMap, storage Storag
 			Msg("Producing weekly notification for ")
 
 		notification := generateWeeklyNotificationMessage(&notificationInsightsAdvisorURL, fmt.Sprint(account), digest)
-		_, _, err := notifier.ProduceMessage(notification)
+		msgBytes, err := json.Marshal(notification)
+		if err != nil {
+			log.Error().Err(err).Msg(invalidJSONContent)
+			os.Exit(ExitStatusMessageEncoderError)
```
Comment (tisnik) [inline differ/differ.go:596]: dtto

File: differ/differ.go [hunk 6]
```diff
@@ -700,14 +840,22 @@ func closeStorage(storage *DBStorage) {
 	}
 }
 
-func closeNotifier() {
-	err := notifier.Close()
+func closeKafkaNotifier() {
+	err := kafkaNotifier.Close()
 	if err != nil {
 		log.Err(err).Msg(operationFailedMessage)
 		os.Exit(ExitStatusKafkaConnectionNotClosedError)
```
Comment (tisnik) [inline differ/differ.go:847]: if it exit, no other close() will be called

File: differ/utils.go [hunk 7]
```diff
@@ -0,0 +1,58 @@
+// Copyright 2022 Red Hat, Inc
```
Comment (tisnik) [inline differ/utils.go:1]: httpUtils perhaps? (it's still bad, but not as bad as `utils` or `common` :)

**Review by tisnik — COMMENTED** [review]
Left some comment

File: differ/utils.go (hunk 7 above)
Comment (JiriPapousek) [reply differ/utils.go:1]: Looked into the insights-operator-utils, could not find these in there.

File: differ/utils.go [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+	// Read body from response
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		log.Error().Msgf("Got error while reading the response's body - %s", err.Error())
+		return nil, err
+	}
+
+	err = response.Body.Close()
+	if err != nil {
+		log.Error().Msgf("Got error while closing the response body - %s", err.Error())
```
[the remark this answers is not in the document]
Comment (JiriPapousek) [reply differ/utils.go:55]: Fixed

File: differ/differ.go (hunk 6 above)
Comment (JiriPapousek) [reply differ/differ.go]: True, added `defer` to `os.Exit` calls.

File: differ/differ.go (hunk 4 above)
Comment (JiriPapousek) [reply differ/differ.go]: Replaced exit with continue

**Pull request closed without merging**